webpack/scenes/ContentViews/Details/Versions/VersionDetails/ContentViewVersionDetailsHeader.js

```diff
@@ -10,52 +11,150 @@ import {
 import EditableTextInput from '../../../../../components/EditableTextInput';
 import { hasPermission } from '../../../helpers';
+import ContentViewVersionPromote from '../../Promote/ContentViewVersionPromote';
+import getEnvironmentPaths from '../../../components/EnvironmentPaths/EnvironmentPathActions';
+import RemoveCVVersionWizard from '../Delete/RemoveCVVersionWizard';
 
 const ContentViewVersionDetailsHeader = ({
   versionDetails: {
-    version, description, environments,
+    version, description, environments, content_view_id: cvId, id,
   },
   onEdit,
   details: { permissions },
-}) => (
-  <>
-    <GridItem span={12}>
-      <TextContent>
-        <Text component={TextVariants.h2}>{__('Version ')}{version}</Text>
-      </TextContent>
-    </GridItem>
-    <GridItem className="content-view-header-content" span={12}>
-      <TextContent>
-        <TextList component={TextListVariants.dl}>
-          <EditableTextInput
-            key={description} // This fixes a render issue with the initial value
-            textArea
-            label={__('Description')}
-            attribute="description"
-            placeholder={__('No description provided')}
-            onEdit={onEdit}
-            disabled={!hasPermission(permissions, 'edit_content_views')}
-            value={description}
-          />
-        </TextList>
-      </TextContent>
-      <Flex>
-        {environments?.map(({ name, id }) =>
-          <FlexItem key={name}><Label isTruncated color="purple" href={`/lifecycle_environments/${id}`}>{name}</Label></FlexItem>)}
-      </Flex>
-    </GridItem>
-  </>
-);
+}) => {
+  const dispatch = useDispatch();
+  useEffect(
+    () => {
+      dispatch(getEnvironmentPaths());
+    },
+    [dispatch],
+  );
+  const [dropdownOpen, setDropdownOpen] = useState(false);
+  const [promoting, setPromoting] = useState(false);
+  const [removingFromEnv, setRemovingFromEnv] = useState(false);
+  const [currentStep, setCurrentStep] = useState(1);
+  const [deleteVersion, setDeleteVersion] = useState(false);
+  const dropDownItems = [
+    <DropdownItem
+      key="promote"
+      onClick={() => {
+        setPromoting(true);
+      }}
+    >
+      {__('Promote')}
+    </DropdownItem>,
+    <DropdownItem
+      key="remove"
+      onClick={() => {
+        setRemovingFromEnv(true);
+      }}
+    >
+      {__('Remove from environment')}
+    </DropdownItem>,
+    <DropdownItem
+      key="delete"
+      onClick={() => {
+        setCurrentStep(1);
+        setDeleteVersion(true);
+        setRemovingFromEnv(true);
+      }}
+    >
+      {__('Delete')}
+    </DropdownItem>,
+  ];
+
+  return (
+    <>
+      <GridItem span={10}>
+        <TextContent>
+          <Text component={TextVariants.h2}>{__('Version ')}{version}</Text>
+        </TextContent>
+      </GridItem>
+      <GridItem span={2} style={{ float: 'right' }}>
+        <Dropdown
+          aria-label="version-action-dropdown"
```

Comment:
Due to the right justification of the following recommendation you will likely need to add: 
   position={DropdownPosition.right}